hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -195,7 +195,7 @@ protected MapProxySupport(String name, MapService service, NodeEngine nodeEngine
         this.partitionService = getNodeEngine().getPartitionService();
         this.lockSupport = new LockProxySupport(new DefaultObjectNamespace(SERVICE_NAME, name),
                 LockServiceImpl.getMaxLeaseTimeInMillis(properties));
-        this.operationProvider = mapServiceContext.getMapOperationProvider(name);
+        this.operationProvider = mapServiceContext.getMapOperationProvider(mapConfig);
```

Comment:
This is the fix: `getMapOperationProvider(String mapName)` gets-or-creates the map container in order to obtain the config and choose the appropriate operation provider. Since the config is already available, it is passed to `getMapOperationProvider(Config config)` and skips the map container lookup.